File: src/main/java/com/diskree/achievetodo/BlockedActionType.java
package com.diskree.achievetodo;

public enum BlockedActionType {

    FOOD("unblock.food"),
    ACTION("unblock.action"),
    BLOCK("unblock.block"),
    TOOLS("unblock.tools"),
    ARMOR("unblock.armor"),
    PORTAL("unblock.portal"),
    VILLAGER("unblock.villager");

    final String unblockPopupTitle;

    BlockedActionType(String unblockPopupTitle) {
        this.unblockPopupTitle = unblockPopupTitle;
    }

    public String getUnblockPopupTitle() {
        return unblockPopupTitle;
    }
}

File: src/main/java/com/diskree/achievetodo/BlockedActionType.kt
package com.diskree.achievetodo

import net.minecraft.text.Text

enum class BlockedActionType(private val unblockPopupTitle: String) {
    FOOD("unblock.food"),
    ACTION("unblock.action"),
    BLOCK("unblock.block"),
    TOOLS("unblock.tools"),
    ARMOR("unblock.armor"),
    PORTAL("unblock.portal"),
    VILLAGER("unblock.villager");

    fun getUnblockPopupTitle(): String {
        return Text.translatable(unblockPopupTitle).string
    }
}
